### PR TITLE
Add SiliconSensor stuff to where drawImage gets called

### PR DESCRIPTION
**Pull request by rmjarvis**
This isn't actually a real pull request, since it hasn't been run, tested, etc.  Rather, it serves as something for @jchiang87  to start from when implementing the B/F and TreeRing stuff with Craig's code.

Still TODO:
* Define treering_center and treering_func for each CCD.
* Write unit tests
* Probably some debugging I would assume

### CODE REVIEW

**Comment by rmjarvis** [comment]
Note: There is a branch called treerings in the imSim repo as well with the corresponding stuff for the sky background addition.

File: python/lsst/sims/GalSimInterface/galSimInterpreter.py [hunk 1]
```diff
@@ -324,18 +349,46 @@ class carrying all of the information for the object whose image
...
+                if self.apply_sensor_model:
+                    sensor = galsim.SiliconSensor(rng=self._rng,
+                                                  treering_center=detector.tree_rings.center,
+                                                  treering_func=detector.tree_rings.func)
+                    surface_ops = [waves, angles]
+                else:
+                    sensor = None
+                    surface_ops = ()
+
+                # Desired position to draw the object.
```
Comment (jchiang87) [inline python/lsst/sims/GalSimInterface/galSimInterpreter.py]: @rmjarvis Here I've updated the code to draw on a postage stamp following your example script.  This speeds up things, of course, but the stamp sizes are sometimes too small for the brighter stars.   Here's an example:
imsim_draw_on_postage_stamp
Any suggestions? A different argument to `getGoodImageSize`?

**Comment by rmjarvis** [comment]
Mm.  That makes sense now that you mention it.  We base the size of the stamp on a maximum fraction of flux allowed to fall off the edge.  The default is 0.5%.

The parameter to adjust this is called folding_theshold in GSParams.  We normally think of it as the fraction of flux folding over from the periodic nature of ffts.  

But in this context, it sets what fraction of the flux is allowed to fall off the edge of the postage stamp.  The default is 0.005, so for bright stars, this will be too small.

I guess you could estimate a good value to use for the bright stars by requiring that the total flux falling off the edge is a small fraction of the sky flux around the edge.  Then set folding_threshold appropriately from that.

One thing to be aware of is that the gsparams parameter needs to be set when first creating the object.  You can't change it after the fact.

**Comment by rmjarvis** [comment]
Proposed sketch of code for setting folding_threshold:
```
# Set folding_threshold so at most one pixel worth of sky flux is allowed to fall off stamp.
folding_threshold = sky_level / flux  

# Don't go above the default 5e-3
folding_threshold = min(folding_threshold, 5e-3)

# Set up a gsparams object to use for constructing this object.
gsparams = galsim.GSParams(folding_threshold=folding_threshold)

# Now use this when constructing the star or galaxy.  E.g.
sigma = np.sqrt(alpha*alpha - eff_pixel_sigma_sq)
gaussian1 = galsim.Gaussian(sigma=sigma, gsparams=gsparams)

sigma = np.sqrt(4.0*alpha*alpha - eff_pixel_sigma_sq)
gaussian2 = galsim.Gaussian(sigma=sigma, gsparams=gsparams)
psf = 0.909*(gaussian1 + 0.1*gaussian2)

point = galsim.DeltaFunction(flux=flux)
star = galsim.Convolve(point, psf)
```
Unfortunately, the way things are structured now, the gsparams need to be set correctly right from the start.  In GalSim 2.0, it will be possible to reset the gsparams after the fact, but not in version 1.5.

You only really need the folding_threshold set for the component with the largest extent.  So for galaxies you could get away with not remaking the PSF each time.  But stars of course are just the PSF, so you need to remake the PSF with the right gsparams.

Also, you only need to remake the PSF if the flux is more than 500 * sky_level.  So probably most of the time, just use the normal PSF object you already have made and stored.  But then for the really bright ones, build a new PSF with the tighter folding_threshold.

**Comment by jchiang87** [comment]
@danielsf I think this is ready to have a serious look along with the treerings PR in the imSim repo.

**Comment by danielsf** [comment]
Okay.  I will look over it.

**Comment by danielsf** [comment]
It looks like this needs to be rebased to `master`.  There are some `afw` API changes that this branch does not include.

**Comment by jchiang87** [comment]
Thanks, Scott.  All of the development work is finished for this branch.   Will you also do the rebase?

**Comment by danielsf** [comment]
Sure.  I will take care of it.

**Comment by danielsf** [comment]
This PR was effectively merged by

https://github.com/lsst/sims_GalSimInterface/commit/d92f13976fc940b1467d68b843ecd6b82b75e889

**Pull request closed without merging**